verifyFiles.py

```diff
@@ -0,0 +1,193 @@
+	list_of_dictionaries_based_on_conditions=[]
+	#study_folders_list : List contains path to the folders contianing 
+	#the study folders based on each condition and os
+	study_folders_list=read_contents_from_file(file_path)
+	for folder in study_folders_list:
+           temp_study_folder_dict=OrderedDict()	   
+	   file_names_and_dir_array=list_files_and_dirs(folder)
+	   temp_study_folder_dict[folder]=retrieve_file_attributes(file_names_and_dir_array,folder)
+	   list_of_dictionaries_based_on_conditions.append(temp_study_folder_dict)
+	return list_of_dictionaries_based_on_conditions
+
+#read_contents_from_file method is used to read the directory path containing the subject folders
+#
+#Input parameter: file containing directory paths as its content
+def read_contents_from_file(file_with_dir_details): 
+# Open the file for reading.
+	with open(file_with_dir_details, 'r') as infile:
+	   data=infile.read()  # Read the contents of the file into memory.
+	   #Return a list of the lines, breaking at line boundaries.
+	   directory_list=data.splitlines()
+	return directory_list
+
+#Method generate_checksum is used for generating checksum of individual files and directories
+#present in each subject folder
+#
+#Input parameters: root directory path , individual file name
+def generate_checksum(root_dir, file_name):
+        hasher=hashlib.md5()
+        if os.path.isfile(os.path.join(root_dir, file_name)):
+            md5_sum=file_hash(hasher,os.path.join(root_dir, file_name))
+	elif os.path.isdir(os.path.join(root_dir, file_name)):
+            md5_sum=directory_hash(hasher,os.path.join(root_dir, file_name))
+	return md5_sum
+
+#Method file_hash is used for generating md5 checksum of a file 
+#
+#Input parameters: file name and hasher
+def file_hash(hasher,file_name):
+    file_content=open(file_name)
+    while True:
+        read_buffer=file_content.read(2**20)
+        if len(read_buffer)==0: 
+	   break
+        hasher.update(read_buffer)
+    file_content.close()
+    return hasher.hexdigest()
+
+#Method directory_hash collects the directory and file names from the directory given as input.
+#Checksum is created on the basis of filenames and directories present in the file input directory.
+#
+#Input parameters: hashed content , path 
+def directory_hash(hasher, dir_path):
+    if os.path.isdir(dir_path):
+        for entry in sorted(os.listdir(dir_path)):
+            hasher.update(entry)
+    return hasher.hexdigest()
+
+#Method generate_common_files_list will create a list containing the common elements from the 
+#different dictionaries corresponding to the conditions in which it was created
+#
+#Input parameters: dictionary with details of files in each study folder , file containing details of directories in each condition
+def generate_common_files_list(study_folder_details_dict_list,file_with_directory_details):
+	common_files_list=[]
+	keys_list=read_contents_from_file(file_with_directory_details)
```

Comment:
I don't think you need to read the list of folders again here. Why don't you just compare the dictionaries in study_folder_details_dict_list?